File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/parsetools/RecordParser.php
<?php 
namespace io\vertx\php\core\parsetools;
use io\vertx\php\core\buffer\Buffer;
use io\vertx\php\core\streams\ReadStream;
class RecordParser
{
    
    private function __construct()
    {

    }
    /**
     * (string delim)
     * @return Void
     * (Buffer delim)
     * @return Void
     * methodSize : 2
     */
    public function delimitedMode(/*1*/$arg0)
    {

    }
    /**
     * (callable endHandler)
     * @return RecordParser
     * methodSize : 1
     */
    public function endHandler(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return RecordParser
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * (int size)
     * @return Void
     * methodSize : 1
     */
    public function fixedSizeMode(/*1*/$arg0)
    {

    }
    /**
     * (Buffer buffer)
     * @return Void
     * methodSize : 1
     */
    public function handle(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return RecordParser
     * methodSize : 1
     */
    public function handler(/*1*/$arg0)
    {

    }
    /**
     * (string delim, callable output)
     * @return RecordParser
     * (string delim, ReadStream stream)
     * @return RecordParser
     * (Buffer delim, callable output)
     * @return RecordParser
     * (Buffer delim, ReadStream stream)
     * @return RecordParser
     * (string delim)
     * @return RecordParser
     * (Buffer delim)
     * @return RecordParser
     * methodSize : 6
     */
    public function newDelimited(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (int size)
     * @return RecordParser
     * (int size, callable output)
     * @return RecordParser
     * (int size, ReadStream stream)
     * @return RecordParser
     * methodSize : 3
     */
    public function newFixed(/*2*/$arg0, $arg1)
    {

    }
    /**
     * ()
     * @return RecordParser
     * methodSize : 1
     */
    public function pause(/*0*/)
    {

    }
    /**
     * ()
     * @return RecordParser
     * methodSize : 1
     */
    public function resume(/*0*/)
    {

    }
    /**
     * (callable output)
     * @return Void
     * methodSize : 1
     */
    public function setOutput(/*1*/$arg0)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/parsetools/RecordParser.java
package io.vertx.php.core.parsetools;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.buffer.Buffer;
import io.vertx.php.core.streams.ReadStream;
@Name("RecordParser")
@Namespace("io\\vertx\\php\\core\\parsetools")
public interface RecordParser<S extends io.vertx.core.parsetools.RecordParser> extends IWrapper<S>{

    @Signature
    default void setOutput(Environment __env__, Memory... args) {
        this.getWrappedObject().setOutput(output)
    }

    @Signature
    static Memory newDelimited(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, RecordParserImpl::new, io.vertx.core.parsetools.RecordParser.newDelimited(delim)
    }

    @Signature
    static Memory newFixed(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, RecordParserImpl::new, io.vertx.core.parsetools.RecordParser.newFixed(size)
    }

    @Signature
    default void delimitedMode(Environment __env__, Memory... args) {
        this.getWrappedObject().delimitedMode(delim)
    }

    @Signature
    default void fixedSizeMode(Environment __env__, Memory... args) {
        this.getWrappedObject().fixedSizeMode(size)
    }

    @Signature
    default void handle(Environment __env__, Memory... args) {
        this.getWrappedObject().handle(buffer)
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }

    @Signature
    default Memory handler(Environment __env__, Memory... args) {
        this.getWrappedObject().handler(handler)
    }
    
    @Signature
    default Memory pause(Environment __env__, Memory... args) {
        this.getWrappedObject().pause()
    }
    
    @Signature
    default Memory resume(Environment __env__, Memory... args) {
        this.getWrappedObject().resume()
    }

    @Signature
    default Memory endHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().endHandler(endHandler)
    }

}
